### ISSUE TITLE
Decimal col type throws Property type mapping mismatch: 

### ISSUE THREAD

**Issue by aszenz** [phpstan/phpstan-doctrine#145]
I have a decimal column in one of the entities, annotating its type with float throws this:

`database can contain string but property expects float.`

I'm not sure what the issue is here, keeping the `type="float"` in `@ORM\Column` doesn't help the same error is throw, what type should i annotate my column with.

Annotating it with `float|int|string` works but I'm not sure its correct

**Comment by ondrejmirtes** [comment]
I'm not sure how this works in PDO, the DecimalType simply returns the value returned by the database driver: https://github.com/doctrine/dbal/blob/37f252326bdc4f83754e114fb64aa0763e73fa48/src/Types/DecimalType.php#L28-L34

So I guess it can really be `float|int|string` depending on some outside factors.

**Comment by aszenz** [comment]
Is a `numeric-string` a better type for Decimal columns?

**Comment by faizanakram99** [comment]
@ondrejmirtes see top comment of the same file

![Image](https://github.com/user-attachments/assets/15982371-a2c0-44ab-8201-1e9ac0b127c5)

https://github.com/doctrine/dbal/blob/37f252326bdc4f83754e114fb64aa0763e73fa48/src/Types/DecimalType.php#L7-L9

**Comment by VincentLanglet** [comment]
> Is a `numeric-string` a better type for Decimal columns?

Yes, you won't have an error since FloatType->toString is a numeric string
https://github.com/phpstan/phpstan-doctrine/blob/27cd48bc59b4913464095288fbf197a66dacb9b8/src/Type/Doctrine/Descriptors/DecimalType.php#L31
